### PR TITLE
nixos/bcachefs: soft-deprecate 'linuxPackages_testing_bcachefs'

### PR DESCRIPTION
**Pull request by Madouura**
## Description of changes
It's time. Somewhat of a follow-up to #267195.
~~@amjoseph-nixpkgs Let me know if you don't want to be co-authored for the ``bcachefs-tools`` commit; Added you to it because this is more or less stealing your PR. Wanted to update everything bcachefs at once.~~
~~``bcachefsEncrypted`` test will not pass until #256638 is out of staging and in master.~~
I am running this PR on my main machine currently, so encryption DOES work.
There *shouldn't* be any truly breaking changes AFAICT.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chayleaf** [comment]
shouldn't it be possible to completely remove the Cargo.lock now that it's checked into git upstream?

**Review by thoughtpolice — APPROVED** [review]
Oh drat, I'm sorry. I'm already running testing and I merged https://github.com/NixOS/nixpkgs/pull/265820 as I wanted `bcachefs-tools`, I didn't realize this PR *also* included it. This otherwise looks great; I'm happy to sign off on it and merge it if you don't mind pulling that one out. Thank you!

**Comment by Madouura** [comment]
> Oh drat, I'm sorry. I'm already running testing and I merged #265820 as I wanted `bcachefs-tools`, I didn't realize this PR _also_ included it. This otherwise looks great; I'm happy to sign off on it and merge it if you don't mind pulling that one out. Thank you!

Done. Went ahead and added an update script and updated to ``1.3.3``.

**Comment by YellowOnion** [comment]
@thoughtpolice  was that PR needed with my PR waiting in staging?

I feel a little bit blind since I'm not being mentioned in some of the maintenance of the code base, especially since I did a lot of work on the PR, I was hoping my PR would get through quicker, maybe I should cherry pick the commits that add me as a maintainer and merge them to master?

@Madouura we probably need to consider snapshots at some point as well, just been sleeping on it till my PR is merged. How did you get it working on your current system, did you endure the recompile times or something else?

Feel free to @ me on NixOS matrix as woobilicious I think most talk is happening in the System Programming area. My sleep schedule is chaotic so I'll try to get reply when I can.

**Comment by Madouura** [comment]
>@Madouura we probably need to consider snapshots at some point as well, just been sleeping on it till my PR is merged. How did you get it working on your current system, did you endure the recompile times or something else?

I'm not sure exactly what you're referencing, but I just have my nixos config flake set to this PR's branch and it works.

**Comment by Madouura** [comment]
>maybe I should cherry pick the commits that add me as a maintainer and merge them to master?

I think that would be the best option. ``linux_bcachefs_testing`` is now deprecated with this PR so I would suggest taking everything except changes to that and maybe the ``util-linux`` changes and put it into master.
IIRC the ``util-linux`` changes are the blocking factor, and the reason I image why it's not in master yet is because of the cryptsetup testing issues.
Speaking of which, when is ``util-linux`` getting official bcachefs support? It's in ``linux-next`` now so I can't imagine it's too far from now...

**Comment by YellowOnion** [comment]
@Madouura  Oh I thought you had my PR running, the ordering of your comments had me thinking that.

There's a few misconceptions you seem to have:

1. `cryptsetup` breaking is a bug triggered by building and running the cryptsetup tests on a bcachefs file system, moving /tmp off bcachefs, or use another system to build it fixed the issue for me.

3. updating `util-linux` **fixes the failing test**, somewhere between 23.05 and 2.11 util-linux was updated, with broken bcachefs UUID support, nixos-generate-config output  started generating  `/dev/disks/by-uuid` paths, but a bug in util-linux *only* detects UUIDs with a freshly formatted FS (when running nixos-generate-config), once the superblock has grown (after installation) `util-linux` fails to read the UUID from the super block and setup `/dev/disks/by-uuid`. That is why the test fails, because the kernel can't find the device, and panics (panic on error).
4.  The changes in `util-linux` and the changes in `bcachefs.nix` are both required for unlocking an encrypted multi-disk system using UUID= syntax, by-uuid symlinks don't work for multi device systems, the current tests fail to account for this scenario, and we have no `boot.initrd.luks.devices.<name>.device` equivalent for bcachefs file systems, the included dynamic script that works directly on the UUID using the patched util-linux, is also far more robust to reboot reordered device paths (something driving me nuts for ages) and adding/removing devices from the bcachefs file system.
6. The `util-linux` patches are official patches that are from the `2.39` branch, they plan on releasing a `.3` release with a few other bug fixes.
7. This PR was pushed in to staging due to the fact that `util-linux` is a system wide dependency, and requires a huge subset of Nixpkgs to be rebuilt.

5. The reason why it works on your system is because you're using a single device, and ran nixos-generate-config  with an old util-linux.

**Comment by Madouura** [comment]
> @Madouura Oh I thought you had my PR running, the ordering of your comments had me thinking that.
> 
> There's a few misconceptions you seem to have:
> 
>     1. `cryptsetup` breaking is a bug triggered by building and running the cryptsetup tests on a bcachefs file system, moving /tmp off bcachefs, or use another system to build it fixed the issue for me.
> 
>     2. updating `util-linux` **fixes the failing test**, somewhere between 23.05 and 2.11 util-linux was updated, with broken bcachefs UUID support, nixos-generate-config output  started generating  `/dev/disks/by-uuid` paths, but a bug in util-linux _only_ detects UUIDs with a freshly formatted FS (when running nixos-generate-config), once the superblock has grown (after installation) `util-linux` fails to read the UUID from the super block and setup `/dev/disks/by-uuid`. That is why the test fails, because the kernel can't find the device, and panics (panic on error).
> 
>     3. The changes in `util-linux` and the changes in `bcachefs.nix` are both required for unlocking an encrypted multi-disk system using UUID= syntax, by-uuid symlinks don't work for multi device systems, the current tests fail to account for this scenario, and we have no `boot.initrd.luks.devices.<name>.device` equivalent for bcachefs file systems, the included dynamic script that works directly on the UUID using the patched util-linux, is also far more robust to reboot reordered device paths (something driving me nuts for ages) and adding/removing devices from the bcachefs file system.
> 
>     4. The `util-linux` patches are official patches that are from the `2.39` branch, they plan on releasing a `.3` release with a few other bug fixes.
> 
>     5. This PR was pushed in to staging due to the fact that `util-linux` is a system wide dependency, and requires a huge subset of Nixpkgs to be rebuilt.
> 
>     6. The reason why it works on your system is because you're using a single device, and ran nixos-generate-config  with an old util-linux.

I appreciate the reply, that answered a few questions/misconceptions I had.

>The reason why it works on your system is because you're using a single device, and ran nixos-generate-config with an old util-linux.

I'm not sure where you got this from. My setup is multi-device and encrypted.
My config also regenerates the (hardware) config every time I run ``make``, so I don't think that's the case, since if it was, I would have been unable to boot several iterations ago.

**Comment by Madouura** [comment]
To be clear, before this PR I actually was running your PR, as a test.

**Comment by YellowOnion** [comment]
@Madouura Oh you must be using colon separated multi device system? I think I incorrectly assumed that wasn't possible on master for some reason. I moved away from that due to devices constantly swapping order and 5 by-id devices being too long for systemd to handle.

Were you using an old version? or perhaps only specific parts of my PR?

I'm now curious why it doesn't break, do you mind sharing?

**Comment by Madouura** [comment]
> @Madouura Oh you must be using colon separated multi device system? I think I incorrectly assumed that wasn't possible on master for some reason. I moved away from that due to devices constantly swapping order and 5 by-id devices being too long for systemd to handle.
> 
> Were you using an old version? or perhaps only specific parts of my PR?
> 
> I'm now curious why it doesn't break, do you mind sharing?

There's not much to say, I cherry-picked your PR's commits into a branch I used to have on my nixpkgs fork, disabled cryptsetup tests, did the correct overlays in my config, and it ran.

**Comment by thoughtpolice** [comment]
> @thoughtpolice was that PR needed with my PR waiting in staging?

Perhaps it was not; I had not checked staging recently, which was my fault &mdash; I instead searched for _open_ bcachefs patches but missed yours since it would have already been merged. I apologize for jumping the gun here! I should have left it to y'all.

> I feel a little bit blind since I'm not being mentioned in some of the maintenance of the code base, especially since I did a lot of work on the PR, I was hoping my PR would get through quicker, maybe I should cherry pick the commits that add me as a maintainer and merge them to master?

Yes, my apologies. Please do add yourself as a maintainer so you can be CC'd on all this!

**Comment by Madouura** [comment]
Staging finally got merged into master. :tada:
All tests pass. Running on my machine. :+1:

**Comment by amjoseph-nixpkgs** [comment]
I'd suggest splitting out the `bcachefs-tools` version-bump into a separate PR.

FWIW personally I'm now using:

- https://github.com/NixOS/nixpkgs/pull/266027

and running bcachefs from `linux-6.6.0-next-20231107`.  It is really, really solid.  So I don't object to this PR but I also won't be likely to test it until `linux_testing` gets past `next-20231107`.

File: nixos/modules/tasks/filesystems/bcachefs.nix
Comment (chayleaf) [inline nixos/modules/tasks/filesystems/bcachefs.nix:101]: It's already in mainline, you should replace the comment with something along the lines of
```suggestion
# Linux 6.7 is required for bcachefs support
# FIXME: replace this with linuxPackages_latest when 6.7 hits stable, remove this line when the LTS version is at least 6.7
```

**Review by chayleaf — COMMENTED** [review]
You should also add a `config.boot.kernelPackages.kernel.kernelAtLeast "6.7"` assertion. And yes, I think splitting bcachefs-tools to a separate PR is a good idea as well.

**Comment by Madouura** [comment]
All tests pass.

**Review by chayleaf — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
Just to be clear, this is not supposed to go before 23.11 branch-off, right? This is a breaking change.

**Comment by chayleaf** [comment]
this is effectively a kernel version bump, is that still a breaking change?

**Comment by RaitoBezarius** [comment]
> this is effectively a kernel version bump, is that still a breaking change?

I see that the 6.4.x series used for bcachefs is removed in this PR and everyone is forced to 6.7-rc1, this is a breaking change given that bcachefs is an experimental (from the perspective of mainline, upstream, etc.) filesystem. There is care and due diligence to do w.r.t. to the `bcachefs-tools`, I personally don't care that we break people's machine that are not necessarily able to recover from that because they just got on the bcachefs hype train, and they will just end up discarding the whole thing and write this as "not ready", but out of respect to Kent Overstreet and the bcachefs developers, I think we ought to be careful in how we do our bidding to minimize the support load on the bcachefs developers.

Therefore, except if people can be certain this won't cause any weird error and render someone unable to boot, let's keep both attributes, **deprecate** it (not **remove it** as it was done artificially here) and then we will remove this just after branch-off. Give people a way to migrate from 23.11 to 24.05.

**Comment by chayleaf** [comment]
In that case, something like this?

```
# FIXME: remove after 23.11 branch-off
kernelPackages = mkOverride 900 linuxPackages_bcachefs;

warnings = lib.optional (config.boot.kernelPackages.kernelOlder "6.7") "linuxPackages_bcachefs is deprecated and will be removed in 24.05, set boot.kernelPackages to linuxPackages_testing to suppress this warning";
```

**Comment by RaitoBezarius** [comment]
> In that case, something like this?
> 
> ```
> # FIXME: remove after 23.11 branch-off
> boot.kernelPackages = lib.mkOverride 500 linuxPackages_bcachefs;
> 
> warnings = lib.optional (config.boot.kernelPackages.kernelOlder "6.7") "linuxPackages_bcachefs is deprecated and will be removed in 24.05, set boot.kernelPackages to linuxPackages_testing to suppress this warning";
> ```

Something like this would be better but again, I am not sure if you migrate from 6.4 to 6.7rc1 this way you will have a proper system. I highly suggest writing a more complicated warning with a link to a tracking issue in nixpkgs on how to migrate from 6.4 to 6.7rc1 and what should be the checklist of things to do.

**Comment by chayleaf** [comment]
well, the tests pass and some people have tested linux_testing on their machines already, plus the patches were always intended to land into mainline some day, so I'm not sure what else there is to do

**Comment by bendlas** [comment]
> so I'm not sure what else there is to do

Maybe just wait for the branch-off?

I get that everyone wants the new stable to benefit from their new work, but the run to cram everything in last minute is creating considerable burden on release managers, for no clear benefit. (e.g. see the debacle around the new postgres version).

We're past the point now, where we should start focussing on fixing breakage, instead of potentially introducing new one. If there is even a sliver of doubt, just start looking at 24.05 instead.

**Comment by RaitoBezarius** [comment]
> well, the tests pass and some people have tested linux_testing on their machines already, plus the patches were always intended to land into mainline some day, so I'm not sure what else there is to do

Well, tests do pass, but I don't think we have a test that tests the migration from the old kernel to the new kernel? If so, please can you write one?

**Comment by Mic92** [comment]
> > so I'm not sure what else there is to do
> 
> Maybe just wait for the branch-off?
> 
> I get that everyone wants the new stable to benefit from their new work, but the run to cram everything in last minute is creating considerable burden on release managers, for no clear benefit. (e.g. see the debacle around the new postgres version).
> 
> We're past the point now, where we should start focussing on fixing breakage, instead of potentially introducing new one. If there is even a sliver of doubt, just start looking at 24.05 instead.

I am not sure I am getting the context correct. Are you against dropping the linux kernel bcachefs package from the next release? This kernel won't receive any updates eventually while we do get new kernel versions backports for the normal kernels.

**Comment by Madouura** [comment]
This may bit a controversial, but personally I think any bcachefs user knows the risk they took when installing it, especially if they installed it as their root partition. This is further supported by how niche bcachefs is.
Bcachefs has always been marked as testing in nixpkgs, which again, is another indicator.
I think it would be better to get everyone on-board with using linux_testing instead ASAP.
Of course, I defer to those who want to avoid breakage, even if I don't think this change is breaking enough to exclude from 23.11.

**Comment by chayleaf** [comment]
yes, in my opinion this is in no way different from previous bcachefs bumps (e.g. 6.4 -> 6.5; now it's 6.5 -> 6.7), and should be handled in a similar way - without considering it a breaking change

**Comment by RaitoBezarius** [comment]
We are 3 days away from branch-off, I would have considered it if it was much earlier, but given the signals I am seeing on the `#bcache@OFTC` channel, I'm not comfortable breaking more things right now regarding bcachefs.

**Comment by bendlas** [comment]
> I am not sure I am getting the context correct. Are you against dropping the linux kernel bcachefs package from the next release?

not necessarily. what I am against is having this conversation 3 days before branch-off ...

> This kernel won't receive any updates eventually

... and the sense of urgency conveyed by sentences like this is the exact reason for that: This makes it way too easy to make mistakes and pile on even more work / create delays / destroy data of member who would otherwise be able to help with hydra failures.

**Comment by Mic92** [comment]
> not necessarily. what I am against is having this conversation 3 days before branch-off ...

I'd would suggest though to backport these changes than anyway after some testing. This is quite common practice with all kinds of kernels and their associated filesystems (zfs). And between branch off and the final release there is enough time. Considering that most bcachefs users are likely to be on a unstable system anyway.

**Comment by YellowOnion** [comment]
> they just got on the bcachefs hype train

As someone who jumped on when it was still powered by steam and coal (with nixos 21.05?) & 4 passengers.

The hard-coded reference to `linux_testing_bcachefs` has always been a pain point for me trying to use newer kernels, I have a permanent mkForce to allow me to run my own version, keeping it around might be *more* of an issue over the next 6 months, especially since the current kernel is only a month old already out of date being a non-lts kernel.

I would prefer completely removing the linux_testing_bcachefs reference ASAP, I highly doubt anyone is *using it externally*, it's only for `filesystems/bcachefs.nix`, I wouldn't even consider this a breaking change since it's not an external API, it's a hacky internal implementation detail.

My only issue is that plenty of Linux kernels prior to 6.7 actually support bcachefs, I would rather the assertion check the kernel config, or be removed to avoid breaking setups that should work. (like my current stable/testing [hybrid jank](https://github.com/YellowOnion/nixos-config/blob/master/purple.nix#L14-L31))

Because it's bcachefs and experimental, IDAF if it breaks things or even being back ported to 23.11 at a later date, as long as the a fix is easy, (like privoding the solution in the error), and I wouldn't know how to turn off an assertion like that, if I kept using kents master on v6.6.

I think @Mic92 's idea of merging after fork makes the most sense with the time window.

Also has anyone seen or heard from @davidak in a while? maybe it's time to remove from maintainers.

**Comment by Madouura** [comment]
@YellowOnion Does this new commit work with your custom setup?
IIRC checking ``structuredExtraConfig`` for ``BCACHEFS_FS`` should be enough.

File: nixos/modules/tasks/filesystems/bcachefs.nix
Comment (chayleaf) [inline nixos/modules/tasks/filesystems/bcachefs.nix:100]:
```suggestion
          lib.elem (kernel.structuredExtraConfig.BCACHEFS_FS or null) [
            lib.kernel.module lib.kernel.yes
          ]
```

Comment (Madouura) [reply nixos/modules/tasks/filesystems/bcachefs.nix]: Didn't see that, give me a minute to change it.

Comment (Madouura) [reply nixos/modules/tasks/filesystems/bcachefs.nix]: Done.

**Comment by Madouura** [comment]
Locally brought back ``linux_testing_bcachefs`` from the dead and changed ``boot.kernelPackages`` back to ``pkgs.linuxPackages_testing_bcachefs`` to test the module changes.
The ``structuredExtraConfig`` assertion/check DOES work, and tests pass for both the local change and his PR's commits.

**Comment by Madouura** [comment]
@YellowOnion Looks like your bcachefs PR broke my config. Getting ``ENOKEY`` when using the previous default of ``/dev/sdXx:/dev/sdZz``.
It's not a hard fix, (Just have to use ``UUID=`` syntax with ``bcachefs show-super /dev/sdXx``'s external UUID but I think I'm going to have to address it in either this PR or another.

**Comment by Madouura** [comment]
We could also just not worry about the ``/dev/sdXx:/dev/sdZz`` issue, and get rid of the ``IFS`` here.
https://github.com/NixOS/nixpkgs/blob/b3e8dae766a3db1f0845a598f71e1267292a470a/nixos/modules/system/boot/stage-1-init.sh#L89-L91

**Comment by RaitoBezarius** [comment]
> > they just got on the bcachefs hype train
> 
> As someone who jumped on when it was still powered by steam and coal (with nixos 21.05?) & 4 passengers.
> 
> The hard-coded reference to `linux_testing_bcachefs` has always been a pain point for me trying to use newer kernels, I have a permanent mkForce to allow me to run my own version, keeping it around might be _more_ of an issue over the next 6 months, especially since the current kernel is only a month old already out of date being a non-lts kernel.

We are not saying to keep the hard-coded reference?

> 
> I would prefer completely removing the linux_testing_bcachefs reference ASAP, I highly doubt anyone is _using it externally_, it's only for `filesystems/bcachefs.nix`, I wouldn't even consider this a breaking change since it's not an external API, it's a hacky internal implementation detail.

No, this API was observed, it's not a hacky internal implementation detail. Nixpkgs, alas, has no API stability, which often means that all API is public and should be stable.

> My only issue is that plenty of Linux kernels prior to 6.7 actually support bcachefs, I would rather the assertion check the kernel config, or be removed to avoid breaking setups that should work. (like my current stable/testing [hybrid jank](https://github.com/YellowOnion/nixos-config/blob/master/purple.nix?rgh-link-date=2023-11-18T08%3A26%3A27Z#L14-L31))

Sure.

> Because it's bcachefs and experimental, IDAF if it breaks things or even being back ported to 23.11 at a later date, as long as the a fix is easy, (like privoding the solution in the error), and I wouldn't know how to turn off an assertion like that, if I kept using kents master on v6.6.

This is not how we do things in nixpkgs, there's a high quality standard of trying very hard at not breaking everyone's setups. We write a bunch of tests for that, we test, etc. If the idea is to have an experimental bcachefs with no carefulness, let's move bcachefs out of tree, this would be more honest.

---

In the meantime, I re-iterate my demands:

- let's merge this with deprecation in mind and leave people's setup on the old 6.4 and put a nasty warning "Please upgrade to 6.7-rc1 or 6.7-rcX whenever it's released (link to some tracking issue in nixpkgs)".
- Offer some migration documentation or troubleshooting documentation in the manual, given what we see again in support channels, this is necessary *out of respect*, Nixpkgs is used by a LOT of people, it's bound to create who will use bcachefs and then go into bcachefs' support channels.
- Offer tests to migrate from old kernel to new kernel to help us to pull the cord on the old kernel (optional, but this is what I would expect someone to do if they wanted to be careful about bcachefs) — this is in general very useful to prevent any surprise bcachefs change as it is an experimental filesystem.

Note that branch-off is in 2 days, if this PR doesn't go through in time, we can backport the relevant contents.

File: pkgs/top-level/aliases.nix
Comment (RaitoBezarius) [inline pkgs/top-level/aliases.nix:562]: This is still 6.7-rc1.

Comment (Madouura) [reply pkgs/top-level/aliases.nix:562]: If you're referencing the test I did earlier, I removed the aliases and all-packages changes to test it.

**Comment by YellowOnion** [comment]
@Madouura theres another pr open that fixes a similar issue using uuids, I really should have setup a testing suite, might do that over the next few days to try and catch corner cases, well probably want to catch any regressions in the future.

**Comment by Madouura** [comment]
>Offer some migration documentation or troubleshooting documentation in the manual, given what we see again in support channels, this is necessary out of respect, Nixpkgs is used by a LOT of people, it's bound to create who will use bcachefs and then go into bcachefs' support channels.

I'm not exactly sure what you want me to put, based off what you're saying we're just keeping what we have now and putting a warning.

**Comment by RaitoBezarius** [comment]
That's for 24.05 and unstable.

**Comment by Madouura** [comment]
>troubleshooting documentation in the manual

Maybe migration documentation for migrating to ``linux_testing`` make sense, but general bcachefs troubleshooting documentation is way out of scope for this PR.
I'm still confused on why though, especially since said migration documentation would basically just be to add ``boot.kernelPackages = lib.mkForce pkgs.linuxPackages_testing``.

File: nixos/modules/tasks/filesystems/bcachefs.nix
Comment (RaitoBezarius) [inline nixos/modules/tasks/filesystems/bcachefs.nix:120]: This should have `lib.mkDefault` ?

File: nixos/tests/installer.nix
Comment (RaitoBezarius) [inline nixos/tests/installer.nix:1031]: No, this should be `system.extraDependencies`

**Comment by Madouura** [comment]
All tests pass. Is this set of commits satisfactory?
```console
/nix/store/xsrjql2kq7wiwr3l8k95q8nwxiqn5gn9-vm-test-run-bcachefs
/nix/store/26cwv970fxslanvwkj7jw2pg42k1cf78-vm-test-run-installer-bcachefs-simple
/nix/store/swglwmhv5l8yp8if33disv75zv2nh9h3-vm-test-run-installer-bcachefs-multi
/nix/store/59l2yyyyqyz2ypxdr2kzkigs203xrams-vm-test-run-installer-bcachefs-encrypted
/nix/store/57d15zga689bi56sndzjgid130c45ldy-vm-test-run-installer-bcachefs-linux-testing
/nix/store/57d15zga689bi56sndzjgid130c45ldy-vm-test-run-installer-bcachefs-linux-testing
/nix/store/57d15zga689bi56sndzjgid130c45ldy-vm-test-run-installer-bcachefs-linux-testing
/nix/store/57d15zga689bi56sndzjgid130c45ldy-vm-test-run-installer-bcachefs-linux-testing
```

File: nixos/modules/tasks/filesystems/bcachefs.nix
Comment (Mic92) [inline nixos/modules/tasks/filesystems/bcachefs.nix:121]:
```suggestion
        lib.warn "Please upgrade to Linux 6.7-rc1 or later: 'linuxPackages_testing_bcachefs' is deprecated. Use 'boot.kernelPackages = pkgs.linuxPackages_testing;' to silence this warning."
```

File: nixos/tests/installer.nix
Comment (Mic92) [inline nixos/tests/installer.nix:1005]: Since this is using `lib.mkDefault` above. Do we need force here?
```suggestion
      boot.kernelPackages = pkgs.linuxPackages_testing;
```

File: nixos/tests/installer.nix
Comment (Mic92) [inline nixos/tests/installer.nix:1000]: Same here.

File: nixos/tests/installer.nix
Comment (Mic92) [inline nixos/tests/installer.nix:1035]: Same question here.

File: nixos/doc/manual/release-notes/rl-2311.section.md
Comment (Mic92) [inline nixos/doc/manual/release-notes/rl-2311.section.md:34]:
```suggestion
  - Please consider changing your NixOS configuration's `boot.kernelPackages` to `linuxPackages_testing` until a stable kernel with bcachefs support is released.
```

File: nixos/tests/installer.nix
Comment (Madouura) [reply nixos/tests/installer.nix]: We shouldn't need to force it AFAIK.

**Comment by Madouura** [comment]
> 283996926-6fd43839-b6d6-42eb-8c7b-8efbdf7da1c0

The error in this image is likely fixed by https://github.com/koverstreet/bcachefs/commit/19cf4dffe8d1c1f8574953f3b7178860056e7eeb.

**Pull request merged**
